app/cdap/components/Connections/Create/reducer.ts

```diff
@@ -303,6 +303,22 @@ function getMapOfConnectorToPluginProperties(allConnectors) {
   return mapOfConnectorPluginProperties;
 }
 
+export function getSelectedConnectorDisplayName(selectedConnector, mapOfConnectorPluginProperties) {
+  if (!selectedConnector) {
+    return;
+  }
+  const pluginKey = `${selectedConnector.name}-${selectedConnector.type}`;
+  const artifactKey = getArtifactKey(selectedConnector.artifact);
+  const widgetJSONKey = `widgets.${pluginKey}`;
+  const selectedWidgetJSON = JSON.parse(
```

Comment:
Is there any way to parse this earlier and store the result? My concern is that this requires parsing the JSON for every name appearance on every render.

Reply:
Now I save the parsed Json into the mapOfConnectorPluginProperties, let me know if this aligns with your thought